api/test-client.js

```diff
@@ -17,6 +17,32 @@ tap.test('connect', t => {
     })
   };
 
+  t.test('supports supplying callback to login', t => {
+    jujulib.connect('wss://1.2.3.4', options, (err, juju) => {
+      t.equal(err, null);
+      t.equal(juju instanceof jujulib.Client, true);
+      t.end();
+    });
+    ws.open();
+  });
+
+  t.test('returns promise, supports no callback', t => {
+    const prom = jujulib.connect('wss://1.2.3.4', options)
+      .then(juju =>
```

Comment:
Move the `.then(juju =>` part on previous line and save one level of indentation? [shrug]